assets/tests/abac_tests.yaml

```diff
@@ -786,4 +786,43 @@ tests:
               user: user:jon
             context:
               "x": -1.79769313486231570814527423731704356798070e+309
-            errorCode: 2000
\ No newline at end of file
+            errorCode: 2000
+  - name: exclusion_resolves_deterministically
+    stages:
+      - model: |
+          model
+            schema 1.1
+
+          type user
+
+          type document
+            relations
+              define admin: [user:*]
+              define viewer: [user with condX] but not admin
+
+          condition condX(x: int) {
+            x < 100
+          }
```

Comment:
Can you also add a test to prove that when the error happens in the right hand side of the `but not`, we also get a deterministic response?

Ie.

```
 model
  schema 1.1
 type user
 type document
    relations
       define admin: [user with condX]
        define viewer: [user:*] but not admin
 condition condX(x: int) {
    x < 100
 }
```

Reply:
Added one here and also a unit style in the `internal/graph/check_test.go` file.

Reply:
The test in this file will only add to the overall test time and it is useless because it's only run once, so you will gain little to no confidence that the fix worked. I'd keep the unit test only.